5-Adversarial-Search/draw.js

```diff
@@ -0,0 +1,313 @@
+		for(let i = 0; i < 9; i++) {
+			this.circle_marks.push(draw_circle((i%3*this.scale/3 - this.scale/3 + x),
+												(Math.floor(i/3)*this.scale/3 - this.scale/3 + y),
+												this.scale,
+												this.group,
+												this.scale))
+			this.cross_marks.push(draw_cross((i%3*this.scale/3 - this.scale/3 + x),
+												(Math.floor(i/3)*this.scale/3 - this.scale/3 + y),
+												this.scale,
+												this.group,
+												this.scale))
+			switch(board.tiles[i]) {
+			case -1: this.cross_marks[i].setAttribute('class', 'gameapp'); break
+			case 0: break
+			case 1: this.circle_marks[i].setAttribute('class', 'gameapp'); break
+			}
+		}
+		/*
+			0 = in progress
+			1 = O won
+			2 = X won
+			3 = draw
+		*/
+		let msg = undefined
+		switch(board.gameState) {
+			case 0: msg = "Draw!"; break
+			case 1: msg = "O won!"; break
+			case 2: msg = "X won!"; break
+			case 3: msg = "Draw!"; break
+		}
+		this.message = draw_text(x, y, msg, 'board_status', canvas)
+		this.group.appendChild(this.message)
+	}
+}
+
+class Highlight {
+	constructor(fill, outline, mark, mask, size, color, opacity){
+		this.content = {}
+		this.fill = fill
+		this.outline = outline
+		this.mark = mark
+		this.mask = mask
+		this.size = size
+		this.color = color
+		this.opacity = opacity
+	}
+
+	SetBoard(id, fill, outline, mark, mask, opacity){
+		this.content[id] = {
+			'fill' : fill,
+			'outline' : outline,
+			'mark' : mark,
+			'mask' : mask,
+			'opacity' : opacity
+		}
+	}
+
+	RemoveBoard(id) {
+		delete this.content[id]
+	}
+
+	GetBoardFill(id, tile) {
+		if (this.content[id] != undefined && this.content[id].mask[tile] == 1)
+			return this.content[id].fill
+		if (this.fill != undefined)
+			return this.fill
+		return 'white'
+	}
+
+	GetBoardOutline(id, tile) {
+		if (this.content[id] != undefined)
+			return this.content[id].outline
+		if (this.outline != undefined)	
+			return this.outline
+		return 'black'
+	}
+
+	GetBoardMark(id, tile) {
+		if (this.content[id] != undefined)
+			return this.content[id].mark
+		if (this.mark != undefined)
+			return this.mark
+		return 'black'
+	}
+
+	SetBranch(from, to, color, size, opacity) {
+		this.content[from + ' ' + to] = {
+			'color' : color,
+			'size' : size,
+			'opacity' : opacity
+		}
+	}
+
+	RemoveBranch(from, to, tile) {
+		delete this.content[from + ' ' + to]
+	}
+
+	GetBranchColor(from, to) {
+		if (this.content[from + ' ' + to] != undefined)
+			return this.content[from + ' ' + to].color
+		if (this.color != undefined)
+			return this.color
+		return 'black'
+	}
+
+	GetBranchSize(from, to) {
+		if (this.content[from + ' ' + to] != undefined)
+			return this.content[from + ' ' + to].size
+		if (this.size != undefined)
+			return this.size
+		return 1
+	}
+
+	GetBranchSize(from, to) {
+		if (this.content[from + ' ' + to] != undefined)
+			return this.content[from + ' ' + to].size
+		if (this.size != undefined)
+			return this.size
+		return 1
+	}
+
+	GetBoardOpacity(id, tile) {
```

Comment:
a little awkward that there is no consistency between method name patterns here ... maybe just use camelCaseEverywhere?